lib/Sema/CodeSynthesis.cpp

```diff
@@ -2048,6 +2048,21 @@ swift::createDesignatedInitOverride(TypeChecker &tc,
   if (superclassCtor->getGenericParams())
     return nullptr;
 
+  // Lookup will sometimes give us initializers that are from the ancestors of
+  // our immediate superclass.  So, from the superclass constructor, we look 
+  // one level up to the enclosing type context which will either be a class 
+  // or an extension.  We can use the type declared in that context to check
+  // if it's our immediate superclass and give up if we didn't.
+  // 
+  // FIXME: Remove this when lookup of initializers becomes restricted to our
```

Comment:
Sounds good, thanks!

Although I think in this case, the problem is that piece of code we saw in NameLookup which checks if the found name was a protocol requirement, and looks up the witness -- I think this needs a special case where if the witness is a constructor, we skip it (since we would have found it otherwise I think)?